tiered-storage/jcloud/src/test/java/org/apache/bookkeeper/mledger/offload/jcloud/StreamingOffloadTest.java

```diff
@@ -0,0 +1,217 @@
+        int i = 0;
+        for (Entry e : cursor.readEntries(10)) {
+            assertEquals(new String(e.getData()), "entry-" + i++);
+        }
+
+        for (Entry e : cursor.readEntries(10)) {
+            assertEquals(new String(e.getData()), "entry-" + i++);
+        }
+
+        for (Entry e : cursor.readEntries(5)) {
+            assertEquals(new String(e.getData()), "entry-" + i++);
+        }
+    }
+
+    @Test
+    public void testSetStreamingOffloadAfterCreated() throws ManagedLedgerException, InterruptedException, IOException,
+            ExecutionException {
+        LedgerOffloader offloader = offloaderGenerator.getOffloader(
+                new HashMap<String, String>() {{
+                    put(TieredStorageConfiguration.MAX_OFFLOAD_SEGMENT_SIZE_IN_BYTES, "1000");
+                    put(offloaderGenerator.getConfig().getKeys(TieredStorageConfiguration.METADATA_FIELD_MAX_BLOCK_SIZE)
+                            .get(0), "5242880");
+                    put(TieredStorageConfiguration.MAX_OFFLOAD_SEGMENT_ROLLOVER_TIME_SEC, "600");
+                    put("offloadMethod", OffloadPolicies.OffloadMethod.LEDGER_BASED.getStrValue());
+                }}
+        );
+        int ENTRIES_PER_LEDGER = 10;
+        ManagedLedgerConfig config = new ManagedLedgerConfig();
+        config.setMaxEntriesPerLedger(ENTRIES_PER_LEDGER);
+        config.setMinimumRolloverTime(0, TimeUnit.SECONDS);
+        config.setRetentionTime(10, TimeUnit.MINUTES);
+        config.setRetentionSizeInMB(10);
+        config.setLedgerOffloader(offloader);
+
+        ManagedLedgerImpl ledger = (ManagedLedgerImpl) factory.open("test_read_and_write", config);
+        int beforeSetStreaming = 1;
+        for (int i = 0; i < beforeSetStreaming; i++) {
+            String content = "entry-" + i;
+            ledger.addEntry(content.getBytes());
+        }
+
+        LedgerOffloader streamingOffloader = offloaderGenerator.getOffloader(
+                new HashMap<String, String>() {{
+                    put(TieredStorageConfiguration.MAX_OFFLOAD_SEGMENT_SIZE_IN_BYTES, "1000");
+                    put(offloaderGenerator.getConfig().getKeys(TieredStorageConfiguration.METADATA_FIELD_MAX_BLOCK_SIZE)
+                            .get(0), "5242880");
+                    put(TieredStorageConfiguration.MAX_OFFLOAD_SEGMENT_ROLLOVER_TIME_SEC, "600");
+                    put("offloadMethod", OffloadPolicies.OffloadMethod.STREAMING_BASED.getStrValue());
+                }}
+        );
+
+        config.setLedgerOffloader(streamingOffloader);
+        ledger.setConfig(config);
+
+        for (int i = beforeSetStreaming; i < ENTRIES_PER_LEDGER * 2.5; i++) {
+            String content = "entry-" + i;
+            ledger.addEntry(content.getBytes());
+        }
+
+        Awaitility.await().atMost(Duration.ofSeconds(30)).untilAsserted(() -> {
+            final LedgerOffloader.OffloadHandle before = ledger.getCurrentOffloadHandle();
+            Thread.sleep(5000);
```

Comment:
You can use `Awaitility.pollInterval` to replace the `sleep`.

Reply:
I tried pullInterval but due to Java Lamba's limited ability it seems I have to use the current way.
```java
        final LedgerOffloader.OffloadHandle currentOffloadHandle = ledger.getCurrentOffloadHandle();
        Awaitility.await().pollDelay(Duration.ofSeconds(5)).pollInterval(Duration.ofSeconds(5)).untilAsserted(() -> {
            final LedgerOffloader.OffloadHandle old = currentOffloadHandle;
            currentOffloadHandle = ledger.getCurrentOffloadHandle();
            Assert.assertSame(old, currentOffloadHandle);
        });
```
<img width="1007" alt="image" src="https://user-images.githubusercontent.com/3381789/106861227-72494f00-6700-11eb-9d68-7fac3a4a4bdc.png">